src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -309,6 +384,10 @@ def _is_node_pipeline_breaker(self, node: LogicalPlan) -> bool:
             # Except and Intersect are pipeline breakers since they are join + distinct
             return True
 
+        if isinstance(node, WithQueryBlock):
+            # WithQueryBlock is a pipeline breaker
```

Comment:
if we are treating the WithQueryBlock as a pipeline breaker, when we are breaking the WtihQueryBlock, we can probably directly remove the WiWtihQueryBlock, but propogate the child up, maybe we can get a ticket to do the improvement in a different pr

Reply:
https://snowflakecomputing.atlassian.net/browse/SNOW-1734385